src/main/java/org/opensearch/security/dlic/rest/api/SecurityConfiguration.java

```diff
@@ -0,0 +1,121 @@
+/*
```

Comment:
Is it possible to add more tests around this class? Could `resourceName` ever be `null` in this class?

Reply:
Does this class represent a single entry in the security index? i.e. A single internal user or a single role?

Reply:
I will look to it one more time, may I can reed of it. I added it since java compiler in JDK 11 can''t resolve `SecurityDynamicConfiguration<?>` type if you use it with `Optional<>` or `ValidationResult<>`

Reply:
I made `SecurityConfiguration` as a companion class.

Reply:
Made `SecurityConfiguration` as a companion class.